src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/BulkUploadOrchestrator.cs

```diff
@@ -173,14 +176,64 @@ private async Task<IEnumerable<UploadError>> GetOverlapErrors(IList<Apprenticesh
                     }
                 }
             }
+           
+            return result;
+        }
+
+        private async Task<IEnumerable<UploadError>> GetEmailOverlapErrors(IList<ApprenticeshipUploadModel> uploadedApprenticeships)
+        {
+            var result = new List<UploadError>();
+            var emailapprentices = new List<Apprenticeship>();
+
+            var j = 0;
+            foreach (var apprentice in uploadedApprenticeships.Where(x =>
+                !string.IsNullOrWhiteSpace(x.ApprenticeshipViewModel.EmailAddress)
+                && x.ApprenticeshipViewModel.StartDate.DateTime.HasValue
+                && x.ApprenticeshipViewModel.EndDate.DateTime.HasValue
+                ))
+            {
+                emailapprentices.Add(new Apprenticeship
+                {
+                    Id = j, //assign a row id, as this value will be zero for files
+                    Email = apprentice.ApprenticeshipViewModel.EmailAddress,
+                    StartDate = apprentice.ApprenticeshipViewModel.StartDate.DateTime.Value,
+                    EndDate = apprentice.ApprenticeshipViewModel.EndDate.DateTime.Value
+                });
+                j++;
+            }
+
+            var emailOverlapRequest = new GetEmailOverlappingApprenticeshipsQueryRequest
+            {
+                Apprenticeship = emailapprentices
+            };
+
+            var emailOverlapResponse = await Mediator.Send(emailOverlapRequest);
+
+            if (emailOverlapResponse != null && emailOverlapResponse.Overlaps.Any())
+            {
+                var validationErrors = uploadedApprenticeships.ToList();
+
+                foreach (var overlapGroup in emailOverlapResponse.Overlaps)
+                {
+                    foreach (var overlap in overlapGroup.OverlappingApprenticeships.ToList())
+                    {
+                        if (string.IsNullOrEmpty(overlap.Apprenticeship.Email)) break;
```

Comment:
again `continue` rather than `break`?

Otherwise the loop won't continue to process the other overlapping apprenticeships

Reply:
done